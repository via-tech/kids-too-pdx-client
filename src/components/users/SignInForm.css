.SignIn {
  label, input {
    display: block;
    margin-top: 1em;
  }
  .error {
    color: red;
  }
}