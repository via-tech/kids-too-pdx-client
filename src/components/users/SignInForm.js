import React from 'react';
import PropTypes from 'prop-types';
import UserInfo from './UserInfo';
import styles from './SignInForm.css';
import { Link } from 'react-router-dom';
import { withError } from '../wrappers/withError';

function SignInForm(props) {
  const { handleSubmit, user, error, history } = props;

  if(error && error.includes('deactivated')) history.push('/activate');

  return (
    <section>
      <form className={styles.SignIn} onSubmit={handleSubmit.bind(null, user)}>

        <UserInfo {...props} />

        <button type="submit">Login</button>
      </form>

      <Link to={'/forgot'}><p>Forgot Password?</p></Link>
    </section>
  );
}

SignInForm.propTypes = {
  handleSubmit: PropTypes.func,
  user: PropTypes.object,
  error: PropTypes.string,
  history: PropTypes.object
};

export default withError(SignInForm);
