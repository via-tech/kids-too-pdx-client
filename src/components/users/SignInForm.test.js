import React from 'react';
import renderer from 'react-test-renderer';
import SignInForm from './SignInForm';
import { MemoryRouter } from 'react-router-dom';

describe('Sign in form', () => {
  it('snapshots SignInForm', () => {
    const props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      user: {
        username: 'theOrg@org.com',
        password: 'pass'
      }
    };

    const tree = renderer.create(
      <MemoryRouter>
        <SignInForm {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots SigninForm with error', () => {
    const props = {
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      error: 'Incorrect username or password'
    };

    const tree = renderer.create(
      <MemoryRouter>
        <SignInForm {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
