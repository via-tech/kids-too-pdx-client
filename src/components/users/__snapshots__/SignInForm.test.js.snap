// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in form snapshots SignInForm 1`] = `
<div
  className="Wrappers"
>
  <section>
    <form
      className="SignIn"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          User Info
        </legend>
        <label>
          Username or Email:
          <input
            name="username"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            onChange={[MockFunction]}
            required={true}
            type="password"
          />
        </label>
      </fieldset>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
    <a
      href="/forgot"
      onClick={[Function]}
    >
      <p>
        Forgot Password?
      </p>
    </a>
  </section>
</div>
`;

exports[`Sign in form snapshots SigninForm with error 1`] = `
<div
  className="Wrappers"
>
  <p
    className="Error"
  >
    Incorrect username or password
  </p>
  <section>
    <form
      className="SignIn"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          User Info
        </legend>
        <label>
          Username or Email:
          <input
            name="username"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
        </label>
        <label>
          Password:
          <input
            name="password"
            onChange={[MockFunction]}
            required={true}
            type="password"
          />
        </label>
      </fieldset>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
    <a
      href="/forgot"
      onClick={[Function]}
    >
      <p>
        Forgot Password?
      </p>
    </a>
  </section>
</div>
`;
